### PR TITLE
Log in as user (and set user session) on Log in as user form submit

### PR DESCRIPTION
**Pull request by nemodjuric**
Connects to #3865 
globaladmin2

### CODE REVIEW

File: client/app/test/TestUsers.jsx [hunk 1]
```diff
@@ -31,6 +32,24 @@ export default class TestUsers extends React.PureComponent {
...
+  };
+
+  userIDOnChange = (value) => this.setState({ userID: value });
+  stationIDOnChange = (value) => this.setState({ stationID: value });
+
+  handleLogInAsUser = () => {
+    this.setState({ isLoggingIn: true });
+    ApiUtil.post(`/test/log_in_as_user?id=${this.state.userID}&station_id=${this.state.stationID}`).then(() => {
+      window.location.reload();
+      this.setState({ isLoggingIn: false });
```
Comment (amprokop) [inline client/app/test/TestUsers.jsx]: This code will probably run before the page reloads. But it's probably not guaranteed. Do we need this line at all, really?

File: client/app/test/TestUsers.jsx [hunk 2]
```diff
@@ -95,19 +114,23 @@ export default class TestUsers extends React.PureComponent {
...
-        {this.props.currentUser.roles.includes('Global Admin') &&
+        { this.props.isGlobalAdmin &&
         <div>
-          <strong>Reproduce user state:</strong>
+          <strong>Log in as user:</strong>
           <TextField
             label="User ID:"
-            name="userID" />
+            name="userID"
+            value={this.state.userID}
```
Comment (amprokop) [inline client/app/test/TestUsers.jsx]: nit: `Id` is more consistent with our style

File: app/controllers/test/users_controller.rb [hunk 3]
```diff
@@ -58,10 +59,22 @@ def index
...
+  def log_in_as_user
+    User.clear_current_user # for testing only
+
+    user = User.find_by(css_id: params[:id], station_id: params[:station_id])
+    return head :not_found if user.nil?
+    session["user"] = user.to_hash
+    session["user"]["id"] = user.css_id
+    session["user"]["name"] = user.full_name
+    session[:regional_office] = user.selected_regional_office ? user.selected_regional_office : user.regional_office
+    head :ok
```
[the remark this answers is not in the document]
Comment (nemodjuric) [reply app/controllers/test/users_controller.rb]: Returns a response that has no content (merely headers). 
head :ok              # 200 OK

File: app/models/user.rb [hunk 4]
```diff
@@ -123,8 +127,8 @@ class << self
     attr_writer :authentication_service
     delegate :authenticate_vacols, to: :authentication_service
 
-    # Empty method used for testing purposes
-    def before_set_user
+    # Empty method used for testing purposes (required)
+    def clear_current_user
```
[the remark this answers is not in the document]
Comment (nemodjuric) [reply app/models/user.rb]: Since it's more explanatory what this method does.

**Comment by mdbenjam** [comment]
Looks good to me. Have we tested in UAT? Would want to make sure it appears.

File: app/controllers/test/users_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (amprokop) [reply app/controllers/test/users_controller.rb]: cool

File: client/app/test/TestUsers.jsx [hunk 5]
```diff
@@ -26,11 +27,27 @@ export default class TestUsers extends React.PureComponent {
...
         console.warn(err);
       });
+  };
+
+  userIdOnChange = (value) => this.setState({ userId: value });
+  stationIdOnChange = (value) => this.setState({ stationId: value });
+
+  handleLogInAsUser = () => {
+    this.setState({ isLoggingIn: true });
+    ApiUtil.post(`/test/log_in_as_user?id=${this.state.userId}&station_id=${this.state.stationId}`).then(() => {
```
Comment (amprokop) [inline client/app/test/TestUsers.jsx]: nit: this is a really long and busy line, and is hard on the eyes.  can we break some of it up?

File: app/controllers/test/users_controller.rb [hunk 6]
```diff
@@ -58,10 +59,22 @@ def index
...
+  end
+
+  def log_in_as_user
+    User.clear_current_user # for testing only
+
+    user = User.find_by(css_id: params[:id], station_id: params[:station_id])
+    return head :not_found if user.nil?
+    session["user"] = user.to_hash
+    session["user"]["id"] = user.css_id
+    session["user"]["name"] = user.full_name
```
Comment (shanear) [inline app/controllers/test/users_controller.rb]: seems like we should just create a method in user called `#to_session_hash`

and then we can write:
```
session["user"] = user.to_session_hash
```

**Pull request merged**